Neos.Neos/Classes/FrontendRouting/SiteDetection/SiteDetectionMiddleware.php

```diff
@@ -42,19 +43,27 @@ final class SiteDetectionMiddleware implements MiddlewareInterface
 
     public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
     {
-        $requestUriHost = $request->getUri()->getHost();
         $site = null;
-        if (!empty($requestUriHost)) {
-            $activeDomain = $this->domainRepository->findOneByHost($requestUriHost, true);
-            if ($activeDomain !== null) {
-                $site = $activeDomain->getSite();
+        $requestUriHost = $request->getUri()->getHost();
+        try {
+            if (!empty($requestUriHost)) {
+                $activeDomain = $this->domainRepository->findOneByHost($requestUriHost, true);
+                if ($activeDomain !== null) {
+                    $site = $activeDomain->getSite();
+                }
             }
-        }
-        if ($site === null) {
-            $site = $this->siteRepository->findFirstOnline();
+            if ($site === null) {
+                $site = $this->siteRepository->findFirstOnline();
+            }
+        } catch (DatabaseException) {
+            // doctrine might have not been migrated yet or not database exists.
+            // this doesn't have to be handled here, and we should allow other middlewares / routes to work
+            return $handler->handle($request);
         }
 
         if (!$site instanceof Site) {
+            // no site has been created yet,
+            // but we allow other middlewares / routes to work
             return $handler->handle($request);
```

Comment:
hmm this might spam the logs in early development but should we log those exceptions?

Reply:
Nah, I don't think this is an actual exception here, we are exploring if there is a site, but at this point it's clearly no error to have none?

Reply:
hmmm yes makes sense ... and what about the doctrine stuff? catch silently?